### PR TITLE
Update Remnant half of the Remnant/Deep PR

### PR DESCRIPTION
**Pull request by Amazinite**
PRing now so that you can see the progress and comment while I'm working.

Updates the Remnant half of the Remnant/Deep PR:
* - [x] Now only one mission that can be done twice instead of four missions with a branch.
* - [x] Conversations branch heavily based on the player's conditions:
  * - [x] On offer:
    * - [x] First time giving information?
       * - [x] Are the Deep scientists only investigating Terminus?
          * - [x] Asked to retrieve information (accept/decline). 
       * - [x] Have the Deep scientists discovered the Ember Waste?
          * - [x] Do the scientists know about the Remnant? (Tests to see if the player told.)
            * - [x] Did the player bring Remnant tech to the Deep?
            * - [x] Did the player bring Ancestral tech to the Remnant?
            * - [x] Did the player plant the cubes?
          * - [x] Asked to retrieve information (accept/decline).
    * - [x] Second time giving information?
       * - [x] We know that the Deep Scientists have discovered the Ember Waste if this is the second time giving information.
          * - [x] Do the scientists know about the Remnant? (Tests to see if the player told.)
            * - [x] Did the player bring Remnant tech to the Deep?
            * - [x] Did the player bring Ancestral tech to the Remnant?
            * - [x] Did the player plant the cubes?
          * - [x] Asked to retrieve information (accept/decline).
  * - [x] On stopover:
     * - [x] Only here for Terminus research?
     * - [x] Here for Ember Waste research?
        * - [x] First time?
        * - [x] Second time?
          * - [x] Did the player tell about Remnant?
          * - [x] Did the player know before or after Terminus?
          * - [x] Did the player bring Remnant technology?
          * - [x] Did the player place the surveillance cubes?
  * - [x] On complete:
     * - [x] Only here for Terminus research?
     * - [x] Here for Ember Waste research?
        * - [x] First time?
        * - [x] Second time?
           * - [x] Did the player get any information?
           * - [x] Did the player get old information?
           * - [x] Did the player get new information?

### CODE REVIEW

**Comment by Zitchas** [comment]
Cool. Thanks!

File: data/remnant/remnant missions.txt [hunk 1]
```diff
@@ -1054,12 +1066,211 @@ mission "Remnant: Deep Surveillance"
...
+				`	"I don't know, I only know that they are."`
+					goto secret
+			
+			apply
+				set "remnant: told about cubes"
+			`	Torza punches a wall, then relaxes. "Thank you for telling us this. We will search for and remove these devices immediately."`
+				goto end
+			
+			label secret
+			`	"Whatever methods they are using, we will discover them in due time," Torza says.`
```
Comment (Zitchas) [inline data/remnant/remnant missions.txt:1190]: I find it very incongruous that the Remnant would not know about the surveillance devices:
a) They have very advanced scanners and it is already established that they regularly do detailed scans of <first>'s ship (for instance, the Broken JD missions)
b) Virtually all Remnant know of <first>, and they stand out in Remnant society as a stranger. There is no chance that <first> "blends into the crowd"
c) Remnant have a strong familiarity with cloaking devices, and have mentioned that smaller things are easier to cloak than larger things. Personal cloaks for surveillance operatives or devices is not unlikely, but a surveillance starling (when <first> strolls out of town) is virtually guaranteed.
d) Remnant are hyper-security conscious. While they may let <first> buy their stuff and be starting to trust them more, it is highly unlikely that they would let them wander off without surveillance. On the one hand, <first> is still a stranger. One that appears trustworthy, but a stranger nontheless. On the other hand, <first> is a valuable asset for gaining close-up knowledge of the rest of the galaxy that they hithertoo have only seen from a distance. It'd be a shame if such a valuable asset was put out of commission by getting lost in a desert or falling off a cliff or into some piece of machinery or something. 

As such, I consider this surprise and outburt of Torza's is either erroneous, or an act intending to deceive the player.

**Comment by Zitchas** [comment]
That is hard to follow. You weren't kidding when you mentioned on Discord that this was the most complex mission/conversation in the game. It's great to have it so that people can look at it and see what is possible... But it sure is intimidating, too.

This interpretation of Torza is definitely making him into a bit of a hothead, prone to spontaneous outburts. I'd like to tweak that a little bit: While it won't necessarily affect what the player sees, I consider those outbursts to be acting. They are specifically calculated actions to convince First of the seriousness of their actions, and to portray himself a certain way. It is to  his advantage if a potential enemy sees him as impetuous and shallow, so the more that First makes themselves look like a potential enemy, the more he'll play up those traits. Inversely, if the player is doing things "right" and keeping Remnant security first and foremost, then they should see more of the calm and calculating side. 

I was generally planning on Torza being a "solid rock" sort of character. He's in charge of Remnant ground forces, and evidently handles quite a bit of the ground security side of things. Not the sort of role that someone impetuous or prone to outbursts is likely to be given. It's his job, and especially once a stranger shows up on their doorstep, I'd expect that he's got twenty different scenarios worked out in his head covering all sorts of variations ranging from "nothing changes" to "... they betray us and hordes of humans and alphas follow in their footsteps."

That being said, if punching a wall and appearing to be impetuous are useful for intimidating, deceiving, or otherwise convincing someone to get back on the correct path... Those can be useful tools.

File: data/remnant/remnant missions.txt (hunk 1 above)
Comment (Amazinite) [reply data/remnant/remnant missions.txt:1190]: I actually thought of doing that whole “pull out the cubes” thing here. 

As far as stripping of the license and hostility goes, that’s something I also thought of doing here, but didn’t want to go that far without a go ahead from you.

**Comment by Amazinite** [comment]
>This interpretation of Torza is definitely making him into a bit of a hothead, prone to spontaneous outburts. I'd like to tweak that a little bit: While it won't necessarily affect what the player sees, I consider those outbursts to be acting.

Something I mentioned on Discord is how Torza may very well be a typically calm character, but looking at it from his perspective, the player has been trusted by the Remnant but is now telling them to their face “hey, I spilled all the beans.” Why wouldn’t someone get angry at that? From the perspective of “they actually knew all along,” though, this being acting makes more sense.

**Comment by Zitchas** [comment]
Yeah, anger is definitely to be expected. Although from Torza's point of view, the actual question here is: "Did you *know* you were spying on us, or are you just incredibly naive?" Neither option is great, really.

This leads to the character getting to pick: (just for the "having placed the cubes" stream)
1) I knowingly betrayed the Remnant and would do it again. (lose *all* licenses, hostile Remnant, kicked off the planet.)
2) I knowingly betrayed the Remnant, but I'm apologetic and want to make amends. (lose capital license, Remnant rep reset to 1
3) I am incredibly naive and ended up betraying the Remnant, so I want to make amends.

I'd be half inclined to really demonstrate the sort of control the Remnant have on their own worlds by doing something like stripping off all Remnant engines & steering, and replacing them with equivalent ionic engines. "You are here at our sufferance, we are not going to let you leave with our prized engines." That might be overboard, but I still kind of want to do it.

File: data/remnant/remnant missions.txt (hunk 1 above)
Comment (Zitchas) [reply data/remnant/remnant missions.txt:1190]: Go for it. I am *all* on board with having Torza confront the player about the cubes, and possibly stripping license(s) depending on how the conversation goes.

**Comment by Amazinite** [comment]
> I'd be half inclined to really demonstrate the sort of control the Remnant have on their own worlds by doing something like stripping off all Remnant engines & steering, and replacing them with equivalent ionic engines. "You are here at our sufferance, we are not going to let you leave with our prized engines." That might be overboard, but I still kind of want to do it.

That would need something like https://github.com/endless-sky/endless-sky/issues/4217 to be feasible.

> [The rest...]

I'll keep the offer conversation as is for now, but if we do want to take the punishment approach then I can go back and add to it after finishing the stopover and completion conversations, which I imagine wouldn't change based off of the offering conversation potentially including punishments.

File: data/remnant/remnant missions.txt [hunk 2]
```diff
@@ -1061,138 +1061,646 @@ mission "Remnant: Learn Sign Follow Up"
...
+				goto knowledge
+			
+			label terminus2
+			choice
+				`	"I helped a group of scientists who were researching the wormhole in Terminus."`
+				`	"I discovered that a group of scientists are researching the wormhole in Terminus."`
+			
+			`	"Do they know how to enter it?" Torza asks, not sounding concerned with this new information.`
+			`	"No," you say. "But they know, or at least think, that it is a wormhole, and are attempting to learn how to enter it."`
+			`	"Well it is unlikely that they will ever learn. The only way to stabilize the wormholes is by using a Key Stone, but I have never heard of one being found in Ancestral space. Learning what these scientists know would still be helpful, though. Would you be able to get us a copy of their findings?"`
```
Comment (Zitchas) [inline data/remnant/remnant missions.txt:1132]: Given that providing a batch of keystones from the Hai is a main mission that the player has likely done at this point, it seems odd that Torza would be unaware of a source of crystals outside the waste. His statement is technically true about Ancestral space, but seems kind of naive nonetheless.

File: data/remnant/remnant missions.txt [hunk 3]
```diff
@@ -1061,138 +1061,646 @@ mission "Remnant: Learn Sign Follow Up"
...
+	# Suspicion == 0: Get everything
+		# surveillance
+	# Suspicion == 1: Get exactly what's asked for no trouble
+		# first time, knew after, brought tech
+	# Suspicion == 2: Get what's asked for, slight suspicion
+		# first time, knew after, no tech || first time, knew before, brought tech
+	# Suspicion == 3: Get old info, suspicious
+		# first time, knew before, no tech || second time, knew before, brought tech
+	# Suspicion == 4: Confronted, get nothing
+		# second time, knew before, no tech
```
Comment (Zitchas) [inline data/remnant/remnant missions.txt:1382]: Thank you for the suspicion numbers cheat-sheet. Very helpful!

File: data/remnant/remnant missions.txt [hunk 4]
```diff
@@ -1061,138 +1061,646 @@ mission "Remnant: Learn Sign Follow Up"
...
+				`	"I promise."`
+					goto digitize
+				`	"I'll try."`
+			`	"Funny," Ivan says with a blank expression.`
+			label digitize
+			`	"Let me just get these folders digitized for you, then I can hand you a data chip with everything on it. It'll only be a few minutes."`
+			apply
+				set "remnant: got deep anomaly research"
+			`	Ivan leaves the room with all the folders, and when he returns he hands you a data chip before shoving all the folders back into the filing cabinet. "Well, anything else you need?"`
+			`	"No. That's all I was here for," you respond. "Good luck on the rest of your work."`
```
Comment (Zitchas) [inline data/remnant/remnant missions.txt:1421]: All this sounds really good. That being said, this point right here seems like a great point to add another small branch in terms of trust. From my perspective, if First shows up, asks bluntly for data, and then leaves; that is suspicious to me. Or at least rude. Either way, something that is likely to reduce Ivan's trust.

On the other hand, if First chats with Ivan about his research and stuff, just happens to ask for a copy of the data in the middle, and then continues chatting some more about the research... That sounds like a natural part of the conversation and something that makes First feel more a part of the team.

File: data/remnant/remnant missions.txt [hunk 5]
```diff
@@ -1061,138 +1061,646 @@ mission "Remnant: Learn Sign Follow Up"
...
+			
+			label technology
+			choice
+				`	"I was wondering what your research on Remnant technology has brought up."`
+				`	"Could I get a copy of your research on Remnant technology?"`
+					goto suspicion
+			branch cubes
+				has "Deep: Remnant Surveillance: done"
+			`	"Ah. Well, there's quite a bit to talk about." Ivan reclines his chair.`
+			`	Ivan pulls out a schematic showing the engines you brought. "The engines you brought us are very interesting. We already mentioned to you how they're rather powerful for their size, more so than even atomic engines of a similar, while not producing very much heat at all. Their major downside though seems to be the immense amount of energy they draw. The energy drain is not surpising given that they seem to be improved ion engines, but the fact that ion engines could ever become more powerful for their size than atomic engines is somethign spectacular. The thrusters also seem to have the ability to provide reverse thrust, making any ship with them rather maneuverable."`
```
Comment (Zitchas) [inline data/remnant/remnant missions.txt:1530]: I think Ivan might be overselling the Remnant engines a bit with that first statement. In terms of space, the closest atomic thruster to the smelter is the A520 which has both more total thrust and more thrust per ton. Although for the Forge class the comparison seems to be other way around, with the A250 having both less thrust and less thrust/ton. Interesting. I guess whether or not Ivan is overselling them basically comes down to which set of engines he got.

File: data/remnant/remnant missions.txt [hunk 6]
```diff
@@ -1061,138 +1061,646 @@ mission "Remnant: Learn Sign Follow Up"
...
+				has "Deep: Remnant Surveillance: done"
+			branch again2
+				"retrieved information" == 2
+			
+			`	"A copy?" Ivan looks off into space for a moment. "Well, I suppose since you helped me so much, I could make an exception for you. I typically don't share findings with those not on my research team until they're finalized. I've been burned one too many times because of that."`
+			`	Ivan leans over to his desk. "This will be for your eyes only."`
+				goto branches
+			
+			label again2
+			`	"Again?" Ivan asks. He leans over to his desk. "I like that you're so interested in my research. I suppose it only makes sense with you helping me, but typically merchant captains aren't this interested in following up on odd jobs they do."`
```
Comment (Zitchas) [inline data/remnant/remnant missions.txt:1554]: This seems like a prime opportunity for the player to say something about how working for Ivan is far from "just some random odd job." After all, it isn't everyday that a merchant captain gets to help with ground breaking research.

File: data/remnant/remnant missions.txt [hunk 7]
```diff
@@ -1061,138 +1061,646 @@ mission "Remnant: Learn Sign Follow Up"
...
+			`	"Aside from fleet movements, the cubes are able to make low quality scans of any ships in the system. While those individual scans are less than what an outfit scan provides, given enough scans you're able to get a clearer picture. Now there were specific ship models that would suddenly disappear from the sensors without a trace. I knew they couldn't have been jumping out of the system because jumping creates a very distinct signature on the sensors. On top of ships disappearing on the sensors, there were also jump signatures with no disceranble source. My only conclusion then is that some of these ships are capable of cloaking themselves.`
+			`	"I actually lost connection to the cubes a few weeks ago now, but I've gathered enough information already."`
```
Comment (Zitchas) [inline data/remnant/remnant missions.txt:1680]: I'm really iffy about how much data they are getting here, and especially how *long* they are getting data. 

If it was less detailed than maybe I could spin it with future missions that the Remnant were "feeding" it data after they found them, but there's far too much reasonably accurate info here for that. 

Given the mix of:
- Remnant security
- Power of Remnant scanners
- In-game missions already make it clear that they are scanning the player's ship whenever they come in.

They should know at the very least what First is bringing to the planet, how big it is, and whether they took it with them them when they left immediately. Then there is First's definite traceability, and the ease with which Remnant locate them all the time, which suggests either they are tracking first, or have someone following them. 

In any case, those sensor cubes should be found within a day, probably more likely within hours. As such, a more likely series of events is that the Remnant locate the devices within hours, so the devices get a few hours of scans; beyond which all the info they collect is being carefully selected and fed by the Remnant. 

As such, I have two suggestions for modifying this section.
1) Have Ivan mention that they are getting an unusual amount of interference, perhaps the ion storms?
2) Remove the part about jump signatures with no discernible source, and rephrase the section to suspect stealth technology of some kind. 

The rest seems reasonable, and  could be interpreted a few different ways; and that ambiguity is good.

Comment (Zitchas) [reply data/remnant/remnant missions.txt:1680]:
```suggestion
			`	"With a cube in each of the Remnant systems, I was able to track all their fleet movements, and I've discovered some interesting things. For one, ships either make trips between the three systems, or they leave the cluster entirely for significant amounts of time, often at least a month at a time. There weren't very many ships or fleets that took short trips out of the cluster, say only one or two weeks in length. Seems like they're sending people on rather lengthy missions.`
			`	"Aside from fleet movements, the cubes are able to make low quality scans of any ships in the system. There seems to be some sort of interference, so individual scans are less than what an outfit scan provides, but given enough scans you're able to get a clearer picture. Now there were specific ship models that would suddenly disappear from the sensors without a trace, usually just as they passed through the shadow of a moon or asteroid, never to reappear. I knew they couldn't have been jumping out of the system because jumping creates a very distinct signature on the sensors. My only conclusion then is that some of these ships are equipped with stealth technology.`
			`	"I actually lost connection to the cubes a few weeks ago now, but I've gathered enough information already."`
```

Comment (Amazinite) [reply data/remnant/remnant missions.txt]: Removed mention of the cloaking/stealth altogether.

File: data/remnant/remnant missions.txt (hunk 6 above)
Comment (Amazinite) [reply data/remnant/remnant missions.txt:1544]: Given that this leads directly into the whole "suspicion" branch, I'd rather not make any changes here, as I think it'd ruin the flow of things.

File: data/remnant/remnant missions.txt (hunk 5 above)
Comment (Amazinite) [reply data/remnant/remnant missions.txt:1520]: I'll need to rework these anyway to make mention of the slowing resistance.

File: data/remnant/remnant missions.txt (hunk 4 above)
Comment (Amazinite) [reply data/remnant/remnant missions.txt:1411]: I'd rather not drag this on for too long. And there's really not much I can think to put after this point. Especially painful would be the face that this endpoint is condensed from various previous branches, meaning that to continue on from here would mean to add more branches to make sure we're talking about the right topic. Either that or we'd need some singular topic that works for any branch prior to this.

File: data/remnant/remnant missions.txt (hunk 5 above)
Comment (Zitchas) [reply data/remnant/remnant missions.txt:1520]: Do they need to make mention of it? The Deep doesn't have any easy access to slowing tech of any kind, so it seems unlikely that they'd be able to identify that ability within it. Perhaps simply a mention of how they have an unusual design or things that they don't understand the purpose of it.

File: data/remnant/remnant missions.txt (hunk 6 above)
Comment (Zitchas) [reply data/remnant/remnant missions.txt:1544]: OK, fair enough.

File: data/remnant/remnant missions.txt (hunk 3 above)
Comment (Amazinite) [reply data/remnant/remnant missions.txt]: Given that I'd constructing the suspicions by adding together various conditions, I can't really take a single scenario and say "this one gets a slight bump or drop" without having even more branching and apply nodes to account for such scenarios. I'd rather just leave things as they are, especially because I wrote Ivan's responses according to these scenarios.

File: data/remnant/remnant missions.txt (hunk 2 above)
Comment (Amazinite) [reply data/remnant/remnant missions.txt:1132]: Should I add mention of Hai space from Torza or leave it as is?

**Review by Zitchas — APPROVED** [review]
This looks pretty good to me. I'm not the best proof-reader, though, so if you are done making changes, I'll merge this so that it becomes part of the main PR on the main repository so that everyone can see it more easily.

File: data/remnant/remnant missions.txt (hunk 5 above)
Comment (Amazinite) [reply data/remnant/remnant missions.txt:1520]: Well they do have the Inhibitor, so they could have shot at a ship that had the engines equipped with the Inhibitor. Although in that case we need to make sure the player also brought the Inhibitor, which they might not have.

**Comment by Amazinite** [comment]
I'm done making changes so long as you have nothing else to comment on that you think I should change.
